Add Controller Navigation

qurious-pixel opened the pull request:
- Apply Rainbow patch
- Add Controller Navigation
- Controller "Back" button opens F1 Menu
- Controller Rumble/Gyro settings at the bottom of the controller menu

![controller_navigation](https://user-images.githubusercontent.com/62252937/171067221-e244b69c-bd52-4dd4-af02-ca2f211d68d6.png)

https://user-images.githubusercontent.com/62252937/171067287-cde79dbd-46dd-423c-aabb-c7fca75cb642.mp4

qurious-pixel committed:
Add Controller Navigation

- Apply Rainbow patch
- Add Controller Navigation
- Controller "Back" button opens F1 Menu
- Controller Rumble/Gyro settings at the bottom of the controller menu

vaguerant commented on libultraship/libultraship/SohImGuiImpl.cpp:
```diff
@@ -722,7 +724,25 @@ namespace SohImGui {
+
+		ImGui::Separator();
+
+                EnhancementCheckbox("Dpad Support on Pause and File Select", "gDpadPauseName");
+                EnhancementCheckbox("DPad Support in Ocarina and Text Choice", "gDpadOcarinaText");
+                EnhancementCheckbox("DPad Support for Browsing Shop Items", "gDpadShop");
```
Looks like you may have forked off a while ago, this reverts back to the old "Dpad/DPad" spelling which was fixed previously in #306.

qurious-pixel replied:
I had pulled an old merge, then corrected, but must have copied the old code. will fix.

GreatArgorath commented on libultraship/libultraship/SohImGuiImpl.cpp:
```diff
@@ -722,7 +724,25 @@ namespace SohImGui {
-            {
+	    {
+		ImGuiIO& io = ImGui::GetIO();
+		ImGui::CheckboxFlags("Use Controller Navigation",     &io.ConfigFlags, ImGuiConfigFlags_NavEnableGamepad);      
```
This doesn't save, relaunching the game resets this setting

qurious-pixel committed:
Restore changes from PR #306

Add back update from https://github.com/HarbourMasters/Shipwright/pull/306

GreatArgorath commented on libultraship/libultraship/SohImGuiImpl.cpp:
```diff
@@ -722,7 +724,27 @@ namespace SohImGui {
-            {
+	    {
+		ImGuiIO& io = ImGui::GetIO();
+		ImGui::CheckboxFlags("Use Controller Navigation",     &io.ConfigFlags, ImGuiConfigFlags_NavEnableGamepad);      
```
Oh also, a tooltip explaining the controls for menu navigation would be great
Something like
`Tooltip("Allows controller navigation of the menu bar\nD-pad to move between items, A to select, and X to grab focus on the menu bar");`
![image](https://user-images.githubusercontent.com/60364512/171448808-cf38f377-b75c-409e-86fe-7a21af78f1a2.png)

qurious-pixel committed:
Add Tooltip of new functionality

Tooltip: 
Allows controller navigation of the menu bar
D-pad to move between items, A to select, and X to grab focus on the menu bar

qurious-pixel committed:
set & save check

qurious-pixel committed:
Fixes for switch port
